src/features/DebugSession.ts

```diff
@@ -1,96 +1,96 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import vscode = require("vscode");
+import vscode from "vscode";
 import {
-    CancellationToken, DebugConfiguration, DebugConfigurationProvider,
-    ExtensionContext, WorkspaceFolder
+    CancellationToken,
+    DebugAdapterDescriptor,
+    DebugAdapterDescriptorFactory,
+    DebugAdapterExecutable,
+    DebugAdapterNamedPipeServer,
+    DebugConfiguration,
+    DebugConfigurationProvider,
+    DebugSession,
+    ExtensionContext,
+    WorkspaceFolder
 } from "vscode";
 import { NotificationType, RequestType } from "vscode-languageclient";
 import { LanguageClient } from "vscode-languageclient/node";
-import { getPlatformDetails, OperatingSystem } from "../platform";
+import { LanguageClientConsumer } from "../languageClientConsumer";
+import { ILogger } from "../logging";
+import { OperatingSystem, getPlatformDetails } from "../platform";
 import { PowerShellProcess } from "../process";
 import { IEditorServicesSessionDetails, SessionManager, SessionStatus } from "../session";
 import { getSettings } from "../settings";
-import { ILogger } from "../logging";
-import { LanguageClientConsumer } from "../languageClientConsumer";
-import path = require("path");
-import utils = require("../utils");
+import path from "path";
+import { checkIfFileExists } from "../utils";
 
 export const StartDebuggerNotificationType =
     new NotificationType<void>("powerShell/startDebugger");
 
 export const StopDebuggerNotificationType =
     new NotificationType<void>("powerShell/stopDebugger");
 
-enum DebugConfig {
+export enum DebugConfig {
     LaunchCurrentFile,
     LaunchScript,
     InteractiveSession,
     AttachHostProcess,
 }
 
+/** Make the implicit behavior of undefined and null in the debug api more explicit  */
+type PREVENT_DEBUG_START = undefined;
+type PREVENT_DEBUG_START_AND_OPEN_DEBUGCONFIG = null;
+type ResolveDebugConfigurationResult = DebugConfiguration | PREVENT_DEBUG_START | PREVENT_DEBUG_START_AND_OPEN_DEBUGCONFIG;
+
+const PREVENT_DEBUG_START = undefined;
+const PREVENT_DEBUG_START_AND_OPEN_DEBUGCONFIG = null;
```

Comment:
Not sure if I mentioned this before, but 👏 I like this a lot more than having to remember undefined vs null in these.